WalletWasabi.Gui/ViewModels/AddressViewModel.cs

```diff
@@ -134,10 +138,40 @@ public bool InEditMode
 			set => this.RaiseAndSetIfChanged(ref _qrCode, value);
 		}
 
+		public string QRImageSavePath
+		{
+			get => _qRImageSavePath;
+			set => this.RaiseAndSetIfChanged(ref _qRImageSavePath, value);
+		}
+
 		public string ExpandMenuCaption => _expandMenuCaption?.Value ?? string.Empty;
 
 		public CancellationTokenSource CancelClipboardNotification { get; set; }
 
+		public async Task SaveQRCodeAsync()
+		{
+			var sfd = new SaveFileDialog();
+
+			sfd.InitialFileName = $"{Address}.png";
+			sfd.Directory = Environment.GetFolderPath(Environment.SpecialFolder.MyPictures);
+			sfd.Filters.Add(new FileDialogFilter() { Name = "Portable Network Graphics (PNG) Image file", Extensions = { "png" } });
+
+			var visualRoot = (ClassicDesktopStyleApplicationLifetime)Application.Current.ApplicationLifetime;
+			var path = await sfd.ShowAsync(visualRoot.MainWindow);
+
+			if (!string.IsNullOrWhiteSpace(path))
+			{
+				var ext = Path.GetExtension(path);
+
+				if (string.IsNullOrWhiteSpace(ext))
```

Comment:
imo if the extension is not `.png` we should add it. For example, i could make a mistake and type `myaddress.pnh` and then, given it has an extension it will be saved with that name. In those cases i think it should be saved as `myaddress.pnh.png`. What do you think @jmacato?

Reply:
I agree, will add that now :+1: